static/app/components/alert.tsx

```diff
@@ -104,92 +172,67 @@ const alertStyles = ({
+`;
+
+const ContentWrapper = styled('div')`
   width: 100%;
 `;
 
-const ExpandContainer = styled('div')`
+const MessageContainer = styled('div')<{
+  showIcon: boolean;
+  showTrailingItems: boolean;
+}>`
+  display: flex;
+  width: 100%;
+  padding-top: ${space(1.5)};
+  padding-bottom: ${space(1.5)};
+  padding-left: ${p => (p.showIcon ? '0' : space(2))};
+  padding-right: ${p => (p.showTrailingItems ? space(1.5) : space(2))};
+`;
+
+const Message = styled('span')`
+  line-height: ${p => p.theme.text.lineHeightBody};
+  position: relative;
+  flex: 1;
+`;
+
+const TrailingItems = styled('div')`
+  height: calc(${p => p.theme.fontSizeMedium} * ${p => p.theme.text.lineHeightBody});
   display: grid;
-  padding-top: ${space(1)};
+  grid-auto-flow: column;
+  grid-template-rows: 100%;
+  align-items: center;
+  gap: ${space(1)};
 `;
 
-const ExpandIcon = styled(props => (
-  <IconWrapper {...props}>{<IconChevron />}</IconWrapper>
-))`
-  transform: ${props => (props.isExpanded ? 'rotate(0deg)' : 'rotate(180deg)')};
-  justify-self: flex-end;
+const TrailingItemsWrap = styled(TrailingItems)`
+  margin-left: ${space(1)};
 `;
 
-const Alert = styled(
-  ({
-    type,
-    children,
-    className,
-    showIcon = false,
-    expand,
-    trailingItems,
-    opaque: _opaque, // don't forward to `div`
-    system: _system, // don't forward to `div`
-    ...props
-  }: AlertProps) => {
-    const [isExpanded, setIsExpanded] = useState(false);
-    const showExpand = defined(expand);
-    const showExpandItems = showExpand && isExpanded;
-
-    const getIcon = () => {
-      switch (type) {
-        case 'warning':
-          return <IconWarning />;
-        case 'success':
-          return <IconCheckmark />;
-        case 'error':
-          return <IconNot />;
-        case 'info':
-        default:
-          return <IconInfo />;
-      }
-    };
-
-    return (
-      <div
-        onClick={() => showExpand && setIsExpanded(!isExpanded)}
-        className={classNames(type ? `ref-${type}` : '', className)}
-        {...props}
-      >
-        {showIcon && <IconWrapper>{getIcon()}</IconWrapper>}
-        <ContentWrapper>
-          <MessageContainer>
-            <StyledTextBlock>{children}</StyledTextBlock>
-            {(showExpand || defined(trailingItems)) && (
-              <TrailingItems>
-                {trailingItems}
-                {showExpand && <ExpandIcon isExpanded={isExpanded} />}
-              </TrailingItems>
-            )}
-          </MessageContainer>
-          {showExpandItems && (
-            <ExpandContainer>
-              {Array.isArray(expand) ? expand.map(item => item) : expand}
-            </ExpandContainer>
-          )}
-        </ContentWrapper>
-      </div>
-    );
-  }
-)<AlertProps>`
-  ${alertStyles}
+const ExpandIcon = styled(IconChevron, {
+  shouldForwardProp: (p: string) => p !== 'isExpanded',
+})<{isExpanded: boolean}>`
+  transform: ${p => (p.isExpanded ? 'rotate(0deg)' : 'rotate(180deg)')};
```

Comment:
👎 use the rotate prop on the icon itself

Reply:
ahh that's from back when we had more styles than just the `transform` - totally didn't notice that we could just remove it now. did so in https://github.com/getsentry/sentry/pull/33356/commits/ca2ea1c137fb47a7e5024d07f1bb334bb3fe2699!